### ISSUE TITLE
3 errors generated

### ISSUE THREAD

**Issue by x5forever** [obfuscator-llvm/obfuscator#85]
In Xcode9.0

CompileC /Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/Objects-normal/arm64/AppDelegate.o test/AppDelegate.m normal arm64 objective-c sudo\ vim\ Obfuscator.xcspec
    cd /Users/x5/Desktop/test
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Users/x5/bin/obfuscator/build/bin/clang -x objective-c -arch arm64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=gnu11 -fobjc-arc -fmodules -fmodules-cache-path=/Users/x5/Library/Developer/Xcode/DerivedData/ModuleCache -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/x5/Library/Developer/Xcode/DerivedData/ModuleCache/Session.modulevalidation -fmodules-validate-once-per-build-session -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Werror=return-type -Wdocumentation -Wunreachable-code -Wno-implicit-atomic-properties -Werror=deprecated-objc-isa-usage -Werror=objc-root-class -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wparentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wuninitialized -Wconditional-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wundeclared-selector -Wno-deprecated-implementations -DDEBUG=1 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -miphoneos-version-min=9.0 -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wstrict-prototypes -Wunguarded-availability -iquote /Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/test-generated-files.hmap -I/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/test-own-target-headers.hmap -I/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/test-all-target-headers.hmap -iquote /Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/test-project-headers.hmap -I/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Products/Debug-iphoneos/include -I/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/DerivedSources/arm64 -I/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/DerivedSources -F/Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Products/Debug-iphoneos -c /Users/x5/Desktop/test/test/AppDelegate.m -o /Users/x5/Library/Developer/Xcode/DerivedData/test-fmxxntuqpyasbeblpydilqnryxlw/Build/Intermediates.noindex/test.build/Debug-iphoneos/test.build/Objects-normal/arm64/AppDelegate.o

While building module 'UIKit' imported from /Users/x5/Desktop/test/test/AppDelegate.h:9:
While building module 'FileProvider' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/System/Library/Frameworks/FileProvider.framework/Headers/FileProvider.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/System/Library/Frameworks/FileProvider.framework/Headers/NSFileProviderDomain.h:71:12: error: attributes may not be specified on a category
@interface NSFileProviderExtension (NSFileProviderDomain)
           ^
While building module 'UIKit' imported from /Users/x5/Desktop/test/test/AppDelegate.h:9:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:64:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:9: fatal error: could not build module 'FileProvider'
#import <FileProvider/NSFileProviderExtension.h>
 ~~~~~~~^
In file included from /Users/x5/Desktop/test/test/AppDelegate.m:9:
/Users/x5/Desktop/test/test/AppDelegate.h:9:9: fatal error: could not build module 'UIKit'
#import <UIKit/UIKit.h>
 ~~~~~~~^
3 errors generated.
Command /Users/x5/bin/obfuscator/build/bin/clang failed with exit code 1

**Comment by x5forever** [comment]
the issue is normally in xcode8, but compile error in xcode9

**Comment by Dendim0n** [comment]
same problem.

**Comment by GJ641102478** [comment]
same problem.

**Comment by sevilS** [comment]
Same problem

**Comment by kennytjw** [comment]
same problem.

**Comment by captain-black** [comment]
same problem.
how to fix it ?

**Comment by hanschen123** [comment]
same problem.how to fix it ?

**Comment by CrabMen** [comment]
same problem ，I have solving this problem for several days. @x5forever @Dendim0n @GJBlog @kennytjw @captain-black Hava you solved this problem？

**Comment by spelle** [comment]
Hi ! :-) Same problem here with a Xcode 9.2 :

While building module 'UIKit' imported from /Users/msc/Documents/o-llvm_sample/o-llvm_sample/ViewController.h:9:
While building module 'FileProvider' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk/System/Library/Frameworks/FileProvider.framework/Headers/FileProvider.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk/System/Library/Frameworks/FileProvider.framework/Headers/NSFileProviderDomain.h:71:12: error: attributes may not be specified on a category
@interface NSFileProviderExtension (NSFileProviderDomain)
           ^
While building module 'UIKit' imported from /Users/msc/Documents/o-llvm_sample/o-llvm_sample/ViewController.h:9:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:64:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.2.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:9: fatal error: could not build module 'FileProvider'
#import <FileProvider/NSFileProviderExtension.h>
 ~~~~~~~^
In file included from /Users/msc/Documents/o-llvm_sample/o-llvm_sample/ViewController.m:9:
/Users/msc/Documents/o-llvm_sample/o-llvm_sample/ViewController.h:9:9: fatal error: could not build module 'UIKit'
#import <UIKit/UIKit.h>
 ~~~~~~~^
3 errors generated.
Command /opt/o-llvm/bin/clang failed with exit code 1

**Comment by xxzizixx** [comment]
same Problem in Xcode 9.4, is there any one had solved this issue?  @x5forever @Dendim0n @GJBlog @kennytjw @captain-black 

While building module 'UIKit' imported from /Users/yangbiao/Desktop/--研究demo--/微信H5支付Demo/微信H5支付Demo/main.m:9:
While building module 'FileProvider' imported from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/FileProvider.framework/Headers/FileProvider.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/FileProvider.framework/Headers/NSFileProviderDomain.h:71:12: error: attributes may not be specified on a category
@interface NSFileProviderExtension (NSFileProviderDomain)
           ^
While building module 'UIKit' imported from /Users/yangbiao/Desktop/--研究demo--/微信H5支付Demo/微信H5支付Demo/main.m:9:
In file included from <module-includes>:1:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIKit.h:64:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.4.sdk/System/Library/Frameworks/UIKit.framework/Headers/NSFileProviderExtension.h:14:9: fatal error: could not build module 'FileProvider'
#import <FileProvider/NSFileProviderExtension.h>
 ~~~~~~~^
/Users/yangbiao/Desktop/--研究demo--/微信H5支付Demo/微信H5支付Demo/main.m:9:9: fatal error: could not build module 'UIKit'
#import <UIKit/UIKit.h>
 ~~~~~~~^
3 errors generated.
Command /Users/yangbiao/Desktop/iOSCodeObfuscation/build/bin/clang failed with exit code 1

**Comment by spelle** [comment]
Hi.

It looks like it is an issue of Objective-C language extensions that are not supported by the compiler.
From my experience, you have two choice : 
- Donwgrade the iOS/MACOS SDK untill finding one without these extensions, or
- Use a more recent version of LLVM that supports these extensions.

I resolved this issue by passing to LLVM-6.0.0 instead of LLVM-4.0.
You can find the port here : https://github.com/spelle/obfuscator/tree/llvm-6.0.0
Keep in mind that this is a work in progress.

Hope this helps.

Cheers.
